src/eckit/io/PooledFile.cc

```diff
@@ -73,10 +74,11 @@ class PoolFileEntry {
 
     void remove(const PooledFile* file) {
         auto s = statuses_.find(file);
-        if (s != statuses_.end()) {
-            statuses_.erase(s);
-        }
+        ASSERT(s != statuses_.end());
```

Comment:
this changes the behaviour slightly, but looks better and ensuring stronger correctness.